src/convnets/densenet.jl

```diff
@@ -1,20 +1,32 @@
-"""
-    dense_bottleneck(inplanes, outplanes; expansion=4)
+function dense_bottleneck(inplanes::Integer, growth_rate::Integer, bn_size::Integer,
+                          dropout_prob)
+    return Chain(cat_channels,
+                 conv_norm((1, 1), inplanes, bn_size * growth_rate;
+                            revnorm = true)...,
+                 conv_norm((3, 3), bn_size * growth_rate, growth_rate;
+                            pad = 1, revnorm = true)...,
+                 Dropout(dropout_prob))
+end
 
-Create a Densenet bottleneck layer
-([reference](https://arxiv.org/abs/1608.06993)).
+function dense_block(inplanes::Integer, num_layers::Integer, bn_size::Integer,
+                     growth_rate::Integer, dropout_prob)
+    layers = [dense_bottleneck(inplanes + (i - 1) * growth_rate, growth_rate, bn_size,
+                                   dropout_prob) for i in 1:num_layers]
```

Comment:
Can't you take this vector and build the nested `Parallel` of a `DenseBlock` recursively? That's how I would try doing it to avoid defining an extra type.

Reply:
I'm afraid I can't quite immediately see that implementation 😅 Could you give me an example?